packages/drawer/src/index.vue

```diff
@@ -0,0 +1,29 @@
+<template>
+  <teleport v-if="appendToBody" to="body">
+    <drawer ref="drawer" v-bind="$attrs">
+      <slot></slot>
+    </drawer>
+  </teleport>
+  <drawer v-else ref="drawer" v-bind="$attrs">
+    <slot></slot>
+  </drawer>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed } from 'vue'
+import Drawer from './drawer.vue'
+
+export default defineComponent({
+  name: Drawer.name,
+  components: { Drawer },
+  setup(props, ctx) {
+    const drawer = ref<typeof Drawer>(null)
+    const appendToBody = computed(() => ctx.attrs['append-to-body'])
+    return {
+      drawer,
+      appendToBody,
+      closeDrawer: () => drawer.value.closeDrawer(),
+    }
+  },
+})
+</script>
```

Comment:
By wrapping `Drawer` like this we ended up with extra overheads which is rendering an additional components (including all lifecycles), I think we should keep `Drawer` minimal when you actually do not need to wrap it up like this, also `functional` components is recommended if you insist doing so.

Reply:
Ok! But the project don't seem to support jsx. Using functional component looks like a painful work. Maybe I 
 have to use a string template for the conditional switch template. Is that a ok resolution?